### PR TITLE
Clean up generate_initializesystem()

### PR DESCRIPTION
**Pull request by hersle**
While working on https://github.com/SciML/ModelingToolkit.jl/pull/3050, I found `generate_initializesystem()` a little hard to follow. I thought I could try to clean it up a little, after it has accumulated several fixes recently.

Please feel free to ignore this if it is not helpful. I know @AayushSabharwal is [working on parameter initialization](https://github.com/SciML/ModelingToolkit.jl/pull/2928), so I wouldn't want this merged before if it clutters that!

## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

### CODE REVIEW

**Comment by AayushSabharwal** [comment]
Thanks, this looks amazing! I can very easily rebase the parameter initialization on top of this, so it shouldn't need to wait.

**Comment by AayushSabharwal** [comment]
Could you rebase this? Would love to get this merged

**Comment by hersle** [comment]
Sure, I will do my best when I get back next week :)

File: src/systems/nonlinear/initializesystem.jl [hunk 1]
```diff
@@ -5,109 +5,83 @@ Generate `NonlinearSystem` which initializes an ODE problem from specified initi
...
+                    # variables specified in u0 overrides defaults
+                    push!(defs, y => x)
                 elseif y isa Symbolics.Arr
-                    # scalarize array # TODO: don't scalarize arrays
-                    _y = collect(y)
-                    for i in eachindex(_y)
-                        push!(filtered_u0, _y[i] => x[2][i])
-                    end
+                    # TODO: don't scalarize arrays
+                    push!(defs, (collect(y) .=> x)...)
```
Comment (ChrisRackauckas) [inline src/systems/nonlinear/initializesystem.jl:48]: Don't splat, instead Symbolics.scalarize

Comment (hersle) [reply src/systems/nonlinear/initializesystem.jl]: 👍

**Comment by hersle** [comment]
I tried to improve and simplify the handling of vector equations and scalarization a bit.
I also added a test for it that fails without these changes, so there are now fixes, too.

**Comment by hersle** [comment]
Tests seem to pass.

**Comment by ChrisRackauckas** [comment]
Is that failure on master?

**Comment by hersle** [comment]
Yes, [looks like it](https://github.com/SciML/ModelingToolkit.jl/actions/runs/11183656661/job/31092709572).

**Review by ChrisRackauckas — APPROVED** [review]
